### PR TITLE
V2 fixes

### PR DESCRIPTION
**Pull request by richard-barrett-madetech**
## Link to JIRA ticket
Trello https://trello.com/b/S4PGTOrF/addresses-api-extension Issue 2, Issue 4 and 'Improvement: Single-line address output'

## Describe this PR
 - Fix the elasticsearch re-indexing issue to ensure there is a 10 minute delay before checking for completion of the re-index process (before switching alias')
 - Add 'SingleLineAddress' to the Address response for V2
 - Correct the use of 'parent_shell' in V2 and rename to 'property_shell'

### *What is the problem we're trying to solve*
 - Fix issues where the SQS message queue was being  continuously written to for the ire-index process (because the time delay for messages wasn't working)
 - Add 'SingleLineAddress' to V2 API output - this feature saves work in client applications
 - Correct the use of 'parent_shell' to include 'shell' properties when set to true (and not when set to false or not present). Also, rename it to 'property_shell' to avoid confusion, as it is not related to property parentage.

### *What changes have we introduced*
 - Re-indexing time delay - code amended to ensure a minimum time delay independent of environment variables specifying the delay. Additional check on time delay added in the lambda invoked by the message queue, to ensure additional messages aren't generated unless there has been a time delay. Unit tests developed around this functionality.
 - SingleLineAddress - new derived attribute on the V2 AddressResponse API response object, code to derive address lines 1 - 4 + town + city on a single line, comma spearated. Unit tests.
 - 'parent_shell'  - removal of the complex code that looked at property parentage and replacement with the more straightforward check on property primary usage. Removal of old unit tests, addition of new unit tests.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,13 +82,15 @@ Or locally if you prefer:
...
 
-If not using make, you should start the test databases in their docker containers before running the unit tests
+If not using make, or to debug the tests in visual studio, start the test databases in their docker containers before starting the tests
 
 ```sh
 docker-compose up -d test-database
 docker-compose up -d test-elasticsearch
 ```
 
+NOTE - if you have a local version of postgres installed (and it is running on the default port 5432), you will need to stop it else the unit tests will fail - the docker postgres also runs on port 5432 and there will be a clash. In windows, go to services and stop the postgres server service.
```
Comment (LBHMKeyworth) [inline README.md:92]: :) this has got me so many times!

**Review by LBHMKeyworth — APPROVED** [review]
All good - had one little thing about a part of a test which seems unnecessary but good other than that.

I take it the SQS delay thing ended up being possible without using Step functions then?

**Comment by richard-barrett-madetech** [comment]
Yes, turned out to be an issue with environment variables and queue settings
Have fixed the if statement - it was wrong, but didn't stop the test
working as when expectedCount is 1 we are not including shells in any of
the test cases. Left in as it works as documentation.

File: AddressesAPI.Tests/V2/E2ETests/SearchAddressIntegrationTests.cs [hunk 2]
```diff
@@ -246,25 +276,43 @@ await TestDataHelper.InsertAddressInDbAndEs(DatabaseContext, ElasticsearchClient
...
+            }
+            else if (expectedCount == 1)
+            {
+                if (includePropertyShell)
+                {
+                    returnedUprns.Should().Contain(flatRecord.UPRN);
+                }
+                else
+                {
+                    returnedUprns.Should().Contain(flatRecord.UPRN);
```
[the remark this answers is not in the document]
Comment (richard-barrett-madetech) [reply AddressesAPI.Tests/V2/E2ETests/SearchAddressIntegrationTests.cs]: Have fixed the if statement - it was wrong, but didn't stop the test working as when expectedCount is 1 we are not including shells in any of the test cases. Left in as it works as documentation.

**Pull request merged**